Explorer/Assets/DCL/UI/LoadingSpinner/UILoadingSpinner.cs

```diff
@@ -0,0 +1,50 @@
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace DCL.UI.LoadingSpinner
+{
+    public class UILoadingSpinner : MonoBehaviour
+    {
+        public Image img;
+        public Color color;
+        public float head;
+        public float tail;
+
+        private Material material;
+
+        private void Start()
+        {
+            GetMaterial();
+        }
+
+        private void GetMaterial()
+        {
+            if (img.maskable) { material = img.materialForRendering; }
```

Comment:
Why have you decided to implement it like this? It's not entirely clear what's happening here.
Moreover, instead of having an update loop in a separate class, you could inherit `UILoadingSpinner` from `Image` itself and override `materialForRendering`

Reply:
This component has been imported from unity-renderer (not my code). Inheritance in this case doesnt feel right, i prefer to keep the composition whenever possible. Also it needs an update since the properties are updated through an animation and later passed to the shader